python/GangaDirac/Lib/Files/DiracFile.py

```diff
@@ -475,51 +475,33 @@ def accessURL(self, thisSE=''):
         if len(self.subfiles) == 0:
 
             files_URLs = self._remoteURLs
-            this_accessURL = ''
+            this_accessURL = {}
 
             for this_SE in files_URLs.keys():
-
-                this_URL = files_URLs.get(this_SE)
-                these_sites_output = execute('getSitesForSE("%s")' % str(this_SE))
-                if thisSE == '':
-                    default_site = execute('getSiteForSE("%s")' % self.defaultSE)
-                else:
-                    default_site = thisSE
-
-                if these_sites_output.get('OK', False):
-                    these_sites = these_sites_output.get('Value')
-                    for this_site in these_sites:
-                        if type(default_site) == type([]):
-                            hasMatched = False
-                            for this_Site_in_SE in default_site:
-                                if this_site == this_Site_in_SE:
-                                    hasMatched = True
-                                    break
-                            if hasMatched:
-                                break
-                        elif type(default_site) == type(''):
-                            if this_site == default_site:
-                                this_accessURL = this_URL
-                                break
-                if this_accessURL != '':
-                    break
-
-            # Cannot find an accessURL
+                this_accessURL[this_SE] = []
+                myurl = execute('getAccessURL("%s" , "%s")' % (self.lfn , str(this_SE)))
+                this_accessURL[this_SE] = myurl['Value']['Successful'][self.lfn]
+            # Cannot find an accessURL. Doesn't really do anything as Dirac throws an error before this point.
             if this_accessURL == '':
+                logger.info( 'Cannot find a replica for the LFN %s' % self.lfn)
                 return []
-
-            # Currently only written to cope with 1 replica per DIRAC file
-            # I think adding multiple accessURL for the same file at 1 site
-            # adds confusion
-            return [this_accessURL]
+            # If the SE isn't specified return a random choice.
+            if thisSE == '':
+              return random.choice(list(this_accessURL.items()))[1]
```

Comment:
This could probably at least be:
```python
return random.choice(list(this_accessURL.values()))
```
but since this is basically getting 'any' entry from the list, we could just grab the first with:
```python
return next(iter(this_accessURL.values()))
```